lib/Twig/Extensions/Node/Trans.php

```diff
@@ -41,6 +41,10 @@ public function compile(Twig_Compiler $compiler)
 
         $function = null === $this->getNode('plural') ? 'gettext' : 'ngettext';
 
+        if (null !== $this->getNode('domain')) {
```

Comment:
this looks wrong to me. You are allowing to pass any expression for the domain, which means that you need to choose between gettext and dgettext at runtime, not at compile time, because the expression could evaluate to ``null``